src/dynamicprompts/parser/parse.py

```diff
@@ -81,18 +82,22 @@ def _configure_wildcard() -> pp.ParserElement:
     return wildcard("wildcard").leave_whitespace()
 
 
-def _configure_literal_sequence(is_variant_literal: bool = False) -> pp.ParserElement:
+def _configure_literal_sequence(
+    braces: tuple[pp.Suppress, pp.Suppress],
+    is_variant_literal: bool = False,
+) -> pp.ParserElement:
     # Characters that are not allowed in a literal
     # - { denotes the start of a variant
     # - # denotes the start of a comment
-    non_literal_chars = r"{#"
+    left_brace, right_brace = braces
+    non_literal_chars = rf"#{left_brace.expr}"
 
     if is_variant_literal:
         # Inside a variant the following characters are also not allowed
         # - } denotes the end of a variant
         # - | denotes the end of a variant option
         # - $ denotes the end of a bound expression
-        non_literal_chars += r"}|$"
+        non_literal_chars += rf"|${right_brace.expr}"
 
     literal = pp.Regex(rf"((?!{double_underscore})[^{non_literal_chars}])+")(
```

Comment:
Now that `non_literal_chars` can be arbitrary user input, it should probably be passed through `re.escape` before adding into a regex.